bin/polykey.ts

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env node
+
+// this will be a CLI entrypoint to this library
+// we expect that this can be used as a CLI
+// application directly as js-polykey
+
+import { program } from 'commander'
+import inquirer from 'inquirer'
+import PolyKey from '../src/Polykey'
+import chalk from 'chalk'
+import os from 'os'
+import fs from 'fs'
+
+let pk: PolyKey
+// Write a `file`
+let tempDir: string
+/*******************************************/
+// initialization
+function initPolyKey(
+    password: string | null,
+    polykeyPath: string | undefined = undefined,
+    publicKeyPath: string | undefined = undefined,
+    privateKeyPath: string | undefined = undefined,
+    privatePassphrase: string | undefined = undefined,
+    verbose: boolean | undefined = undefined
+) {
+    try {
+        // Initialize polykey
+        if (password !== null) {
+            pk = new PolyKey(Buffer.from(password), undefined, undefined, polykeyPath)
+        } else if (process.env.PK_KEY !== undefined) {
+            pk = new PolyKey(Buffer.from(process.env.PK_KEY!), undefined, undefined, polykeyPath)
+        } else {
+            console.log(chalk.red('failed to initialize polykey: password was null'));
+        }
+        // import keys if provided
+        if (publicKeyPath !== undefined) {
+            pk.km.loadPublicKey(publicKeyPath)
+            if (verbose) {
+                console.log(chalk.green('public key successfully imported'))
+            }
+        }
+        if (privateKeyPath !== undefined) {
+            if (privatePassphrase !== undefined) {
+                try {
+                    pk.km.loadPrivateKey(privateKeyPath)
+                    if (verbose) {
+                        console.log(chalk.green('private key successfully imported'))
+                    }
+                } catch (err) {
+                    console.log(chalk.red(`Failed to import private key: ${err.message}`))
+                }
+            } else {
+                console.log(chalk.red('passphrase for private key not provided'))
+            }
+        }
+        if (verbose) {
+            console.log(chalk.green(`PolyKey was initialized successfully at '${pk.polykeyPath}'`));
+        }
+        // Initialization of temp directory (for testing)
+        tempDir = fs.mkdtempSync(`${os.tmpdir}/polykeytest`, undefined).toString()
+        fs.writeFileSync(`${tempDir}/file`, Buffer.from('I am to be signed'))
+    } catch (err) {
+        console.log(chalk.red(`Failed to initialize polykey: ${err.message}`))
+    }
+}
+// Run initialization
+/*******************************************/
+const polykey = new program.Command()
+polykey
+    .version(require('../package.json').version, '--version', 'output the current version')
+
+/*******************************************/
+// init
+const config = polykey.command('config')
+    .description('configure polykey')
+    .option('--password <password>', 'provide the password to polykey')
```

Comment:
Do these provide short options as well? See Python's argparse, we should have both short options and long options.

Reply:
I have added a short syntax for every option